src/python/WMCore/MicroService/DataStructs/Workflow.py

```diff
@@ -278,13 +294,13 @@ def getChunkBlocks(self, numChunks=1):
           * and a list integers, which references the total size of each chunk in
             the list above (same order).
         """
-        if numChunks <= 1:
+        if numChunks == 1:
             thisChunk = set()
             thisChunk.update(self.getPrimaryBlocks().keys())
-            thisChunkSize = sum(self.getPrimaryBlocks().values())
+            thisChunkSize = sum([blockInfo['blockSize'] for blockInfo in self.getPrimaryBlocks().values()])
```

Comment:
Here I suggest to make it safe and replace `blockInfo["blockSize"]` with `blockInfo.get("blockSize", 0)` and even better would be to add if statement to test that blockInfo really exists and is a dict type.

Reply:
I think this can be guaranteed from the origin of this information (getBlockReplicasAndSize)